test/unit/StakingPool/depositTo.js

```diff
@@ -0,0 +1,774 @@
+    expect(deposit.lastAccNxmPerRewardShare).to.equal(0);
+    expect(deposit.stakeShares).to.equal(newStakeShares);
+    expect(deposit.rewardsShares).to.equal(newRewardShares);
+  });
+
+  it('register deposit to an existing nft', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    // first deposit
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
+      },
+    ]);
+
+    const firstDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+
+    const newStakeShares = await estimateStakeShares({ amount, stakingPool });
+
+    // deposit to the same tokenId
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: depositNftId,
+        destination,
+      },
+    ]);
+
+    const updatedDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+    const newRewardShares = await getNewRewardShares({
+      stakingPool,
+      initialStakeShares: firstDepositData.stakeShares,
+      stakeSharesIncrease: newStakeShares,
+      initialTrancheId: firstActiveTrancheId,
+      newTrancheId: firstActiveTrancheId,
+    });
+    expect(updatedDepositData.pendingRewards).to.equal(0);
+    expect(updatedDepositData.lastAccNxmPerRewardShare).to.equal(0);
+    expect(updatedDepositData.stakeShares).to.equal(firstDepositData.stakeShares.add(newStakeShares));
+    expect(updatedDepositData.rewardsShares).to.equal(firstDepositData.rewardsShares.add(newRewardShares));
+  });
+
+  it('updates deposit pendingRewards and lastAccNxmPerRewardShare', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    // first deposit
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
+      },
+    ]);
+
+    const depositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+
+    expect(depositData.lastAccNxmPerRewardShare).to.equal(0);
+    expect(depositData.pendingRewards).to.equal(0);
+
+    // Generate rewards
+    const coverRequest = {
+      coverId: 0,
+      productId: 0,
+      amount: parseEther('100'),
+      period: 3600 * 24 * 30, // 30 days
+      gracePeriod: 3600 * 24 * 30,
+      globalCapacityRatio: 20000,
+      capacityReductionRatio: 0,
+      rewardRatio: 10000,
+    };
+    await stakingPool.connect(this.coverSigner).allocateStake(coverRequest);
+
+    await increaseTime(
+      20 * // days
```

Comment:
let's use a function for this calculation

we have daysToSeconds defined in a couple of places 

can move it to.a common utils file as well and import from there

Reply:
Thanks! Found that method in the libs, updated the code to use it [Use daysToSeconds helper method](https://github.com/NexusMutual/smart-contracts/pull/495/commits/3fdabc6f56118807ad09e24e29fbf9a7c731f604)